onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -560,6 +569,20 @@ static bool IsMovingSingleAxis(const std::vector<size_t>& permutations, size_t&
   return single_axis_moved;
 }
 
+bool IsReshape(const std::vector<size_t>& perm, const std::vector<int64_t>& input_dims) {
```

Comment:
This function can be tested individually. It would be better if there are some unit tests for it. 
And, what if `perm` contains invalid values? Can you trust the input?

Reply:
Is there an existing function that can be reused at there?

Reply:
I would need to expose that function in the header then.

Reply:
fwiw in the internal usage the values of perm are validated earlier. by making it public it changes what could in theory be passed (potentially leading to someone adding unnecessary code to re-validate perm for the theoretical but non-existent use-case).